File: frontend/src/services/agenda.ts
import { fetchJson } from '@/lib/api';

export interface EventSession {
    id: number;
    event_id: number;
    title: string;
    description?: string;
    start_time: string;
    end_time: string;
    location?: string;
    speaker_name?: string;
    created_at: string;
    updated_at: string;
}

export interface EventSessionCreate {
    title: string;
    description?: string;
    start_time: string;
    end_time: string;
    location?: string;
    speaker_name?: string;
    event_id: number;
}

export interface EventSessionUpdate {
    title?: string;
    description?: string;
    start_time?: string;
    end_time?: string;
    location?: string;
    speaker_name?: string;
}

export const AgendaService = {
    getEventSessions: async (eventId: number): Promise<EventSession[]> => {
        return fetchJson<EventSession[]>(`/api/v1/events/${eventId}/sessions`);
    },

    createSession: async (eventId: number, data: EventSessionCreate): Promise<EventSession> => {
        return fetchJson<EventSession>(`/api/v1/events/${eventId}/sessions`, {
            method: 'POST',
            body: JSON.stringify(data)
        });
    },

    updateSession: async (sessionId: number, data: EventSessionUpdate): Promise<EventSession> => {
        return fetchJson<EventSession>(`/api/v1/sessions/${sessionId}`, {
            method: 'PUT',
            body: JSON.stringify(data)
        });
    },

    deleteSession: async (sessionId: number): Promise<void> => {
        await fetchJson<void>(`/api/v1/sessions/${sessionId}`, {
            method: 'DELETE'
        });
    }
};

File: backend/app/schemas/agenda.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

# Shared properties
class EventSessionBase(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    speaker_name: Optional[str] = None

# Properties to receive on creation
class EventSessionCreate(EventSessionBase):
    title: str
    start_time: datetime
    end_time: datetime
    event_id: int

# Properties to receive on update
class EventSessionUpdate(EventSessionBase):
    pass

# Properties shared by models stored in DB
class EventSessionInDBBase(EventSessionBase):
    id: int
    event_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# Properties to return to client
class EventSession(EventSessionInDBBase):
    pass
